WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -373,6 +373,16 @@ public void onEventMainThread(PostEvents.PostUploadStarted event) {
         }
     }
 
+    /*
+    * Upload cancelled (probably due to failed media), reload so correct status on uploading post appears
+    */
+    @SuppressWarnings("unused")
+    public void onEventMainThread(PostEvents.PostUploadCanceled event) {
+        if (isAdded() && mSite.getId() == event.localSiteId) {
```

Comment:
Done in f18488f (I also made `updateSiteOrFinishActivity()` get called first, which should prevent any events being handled before we've finished setting up `mSite`).